client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -41,20 +46,96 @@ export class TeaListContainer extends React.Component {
 
       const revSortOrder = [...newSortOrder].reverse();
 
-      currentState = {
+      currentSortState = {
         columnName: columnName,
         asc: newSortOrder,
         desc: revSortOrder
       };
     }
     this.setState({
       sortedIDs: newSortOrder,
-      memoizedIDs: currentState,
+      memoizedIDs: currentSortState,
```

Comment:
You probably don't need to re-set `memoizedIds` unless `currentSortState` is actually changed.

Maybe consider creating a `newState` object that contains properties that will be passed into this `setState` call, which means that on line 49 you could go `newState.memoizedIds = {...` and avoid passing the existing value by default.